client/src/Components/Home/index.jsx

```diff
@@ -1,11 +1,209 @@
+    if (filterQuery === 'Members') {
+      this.setState({ filterMembers: filterSkills(members, skills) });
+    }
+    if (filterQuery === 'Offers') {
+      const filteredOffersSkills = filterSkills(offers, skills);
+      const filtereOffersOfferTypes = filterOfferTypes(offers, offerTypes);
+
+      filteredOffers = filteredOffersSkills.filter(item => {
+        return filtereOffersOfferTypes.filter(_item => item.id !== _item.id);
+      });
+      this.setState({ filteredOffers });
+    }
+  };
+
+  handleOfferTypeOnChange = offerTypes => {
+    let filteredOffers = [];
+    // make a patch request to the back that add new values to filter
+    this.setState({ offerTypes });
+    const { offers, skills } = this.state;
+    const filteredOffersSkills = filterSkills(offers, skills);
+    const filtereOffersOfferTypes = filterOfferTypes(offers, offerTypes);
+
+    filteredOffers = filteredOffersSkills.filter(item =>
+      filtereOffersOfferTypes.filter(_item => item.id === _item.id)
+    );
+    this.setState({ filteredOffers });
+  };
+
+  render() {
+    const {
+      isClicked,
+      skills,
+      offerTypes,
+      filteredOffers,
+      filterMembers,
+    } = this.state;
+    // eslint-disable-next-line react/prop-types
+    return (
+      <>
+        <Row className="home__contanier">
+          <Col className="home__filter" sm={12} lg={3} md={3}>
+            <AutoCompleteTags
+              type="skill"
+              data={skills}
+              onchange={this.handleSkillOnChange}
+            />
+            <br />
+            {localStorage.getItem('filterQuery') === 'Offers' && (
+              <AutoCompleteTags
+                type="offer_type"
+                data={offerTypes}
+                onchange={this.handleOfferTypeOnChange}
+              />
+            )}
+          </Col>
+          <Col className="home__main" sm={12} lg={8} md={8}>
+            <Row>
+              <Col xs={7}>
+                <InputGroup className="mb-2  home__search">
+                  <InputGroup.Prepend>
+                    <Button
+                      variant="outline-secondary"
+                      className="home__search-btn"
+                    >
+                      <i className="fas fa-search home__search-icon" />
+                    </Button>
+                  </InputGroup.Prepend>
+                  <FormControl aria-describedby="basic-addon1" />
+                </InputGroup>
+              </Col>
+              <Col className="home__result-label" xs={2}>
+                {localStorage.getItem('filterQuery') === 'Members'
+                  ? filterMembers.length
+                  : filteredOffers.length}{' '}
+                results
+              </Col>
+              <Col className="dropdown-toggled" xs={3}>
+                {isClicked ? (
+                  <Dropdown.Menu show className="dropdwon-list">
+                    <Dropdown.Header
+                      onClick={() => {
```

Comment:
@Fatmasiam For this click handler, and the ones on line 171 and 186, it would improve performance if you refactored them into class functions (similar to your handleOfferTypeOnChange function).  Then here you could just write `onClick={this.setFilterQueryToOffers}`
This is better than having an arrow function inside your render, because arrow functions inside render will be recreated every time render happens, which results in a lot of unnecessary copies of that function being created in memory.

Reply:
@AymanAlqoqa please do this one